packages/vc.js/src/vc-ld/__tests__/Ed25519Signature2018.transmute/__fixtures__/dereferencingDocumentLoader.ts

```diff
@@ -0,0 +1,78 @@
+import { contexts } from "./contexts";
+import controller from "./controller.json";
+
+import jsonld from "jsonld";
+
+const contextResolver = async (iri: string) => {
+  if (contexts[iri]) {
+    return { document: contexts[iri] };
+  }
+  return undefined;
+};
+
+const documentResolver = async (iri: string) => {
+  if (iri.startsWith(controller.id)) {
+    return {
+      documentUrl: controller.id,
+      document: controller
+    };
+  }
+  return undefined;
+};
+
+// Note: you implement this without using JSON-LD frame,
+// but it MUST return a document with an @context.
+const documentDereferencer = async (document: any, iri: string) => {
+  try {
+    const frame = await jsonld.frame(
```

Comment:
again the use of frame makes me concerned... prefer to see it removed.

Reply:
I will look at removing the frame!

Reply:
This has been removed!